lib/vanagon/project.rb

```diff
@@ -476,5 +476,33 @@ def generate_package
     def generate_packaging_artifacts(workdir)
       @platform.generate_packaging_artifacts(workdir, @name, binding)
     end
+
+    # Generate a json file which lists all of the dependant components of the project
+    #
+    # @return [Hash] where the top level keys are components and their values
+    #   are hashes with additional information on the component.
+    def generate_dependencies_info
+      all_components = {}
+      @components.each do |component|
+        all_components = all_components.merge(component.get_dependency_hash)
+      end
+      all_components
+    end
+
+    # Writes a json file at `ext/build_metadata.json` containing information
+    # about what went into a built artifact
+    def generate_manifest_json
+      manifest = {
+        "packaging_type" => {
+          "vanagon" => Gem.loaded_specs["vanagon"].version.to_s,
+        },
+        "git_sha" => version,
+        "components" => generate_dependencies_info,
+        "build_time" => BUILD_TIME.iso8601,
```

Comment:
We used that other time format in the corresponding ezbake changes.

I'm wondering if it will be awkward to use different formats to represent the same thing.

Not sure.